packages/jest-haste-map/src/lib/getPlatformExtension.js

```diff
@@ -28,7 +28,8 @@ function getPlatformExtension(
   }
   const platform = file.substring(secondToLast + 1, last);
   // If an overriding platform array is passed, check that first
-  if (platforms && platforms.includes(platform)) {
+
```

Comment:
Was waiting until I had the time to get around to it. Definitely agree on the value of backwards compatibility.